src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -323,6 +296,29 @@ create_reference_table(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * EnsureTableCanBeCreated checks if
+ * - we are on the coordinator
+ * - Citus version is correct
+ * - the current user is the owner of the table
+ * - relation kind is supported
+ */
+static void
+EnsureTableCanBeCreated(Oid relationOid)
```

Comment:
Even we do not intercept creation of regular PostgreSQL tables, I would prefer `EnsureCitusTableCanBeCreated`.